crates/nu-command/src/system/run_external.rs

```diff
@@ -537,6 +537,7 @@ impl ExternalCommand {
             .to_string();
 
         let mut process = std::process::Command::new(head);
+        process.env_clear();
```

Comment:
Not sure if we can completely nuke the environment on Windows:
- https://github.com/rust-lang/rust/issues/114737
- https://github.com/nushell/nushell/pull/9976#issuecomment-1675035575

Are we modifying the env later before spawning?

Reply:
> Are we modifying the env later before spawning?

Yep, of course. So if the necessary windows environment variables are present in nushell's emulated env, then the child process will get them. But if a user uses `hide-env` on one of those special env vars then :shrug:

Reply:
As this method is `pub` maybe worth documenting. The only outside use in `exec` seems to also properly set the environment.